diff -Naurp memcache-3.0.6/memcache.c memcache-3.0.6.oden/memcache.c
--- memcache-3.0.6/memcache.c	2012-05-04 09:55:52.630821705 +0200
+++ memcache-3.0.6.oden/memcache.c	2012-05-04 09:58:18.791796654 +0200
@@ -718,7 +718,7 @@ mmc_t *mmc_find_persistent(const char *h
 			mmc_server_free(mmc TSRMLS_CC);
 			mmc = NULL;
 		} else {
-			zend_list_insert(mmc, le_memcache_server);
+			zend_list_insert(mmc, le_memcache_server TSRMLS_CC);
 		}
 	}
 	else if (le->type != le_memcache_server || le->ptr == NULL) {
@@ -735,7 +735,7 @@ mmc_t *mmc_find_persistent(const char *h
 			mmc = NULL;
 		}
 		else {
-			zend_list_insert(mmc, le_memcache_server);
+			zend_list_insert(mmc, le_memcache_server TSRMLS_CC);
 		}
 	}
 	else {
@@ -775,7 +775,7 @@ static mmc_t *php_mmc_pool_addserver(
 	if (zend_hash_find(Z_OBJPROP_P(mmc_object), "connection", sizeof("connection"), (void **)&connection) == FAILURE) {
 		pool = mmc_pool_new(TSRMLS_C);
 		pool->failure_callback = &php_mmc_failure_callback;
-		list_id = zend_list_insert(pool, le_memcache_pool);
+		list_id = zend_list_insert(pool, le_memcache_pool TSRMLS_CC);
 		add_property_resource(mmc_object, "connection", list_id);
 	}
 	else {
@@ -836,7 +836,7 @@ static void php_mmc_connect(INTERNAL_FUN
 		int list_id;
 		mmc_pool_t *pool = mmc_pool_new(TSRMLS_C);
 		pool->failure_callback = &php_mmc_failure_callback;
-		list_id = zend_list_insert(pool, le_memcache_pool);
+		list_id = zend_list_insert(pool, le_memcache_pool TSRMLS_CC);
 		mmc_object = return_value;
 		object_init_ex(mmc_object, memcache_ce);
 		add_property_resource(mmc_object, "connection", list_id);
@@ -986,7 +986,7 @@ static void php_mmc_failure_callback(mmc
 
 	/* check for userspace callback */
 	if (param != NULL && zend_hash_find(Z_OBJPROP_P((zval *)param), "_failureCallback", sizeof("_failureCallback"), (void **)&callback) == SUCCESS && Z_TYPE_PP(callback) != IS_NULL) {
-		if (IS_CALLABLE(*callback, 0, NULL)) {
+		if (MEMCACHE_IS_CALLABLE(*callback, 0, NULL)) {
 			zval *retval = NULL;
 			zval *host, *tcp_port, *udp_port, *error, *errnum;
 			zval **params[5];
@@ -1213,7 +1213,7 @@ PHP_FUNCTION(memcache_add_server)
 	}
 
 	if (failure_callback != NULL && Z_TYPE_P(failure_callback) != IS_NULL) {
-		if (!IS_CALLABLE(failure_callback, 0, NULL)) {
+		if (!MEMCACHE_IS_CALLABLE(failure_callback, 0, NULL)) {
 			php_error_docref(NULL TSRMLS_CC, E_WARNING, "Invalid failure callback");
 			RETURN_FALSE;
 		}
@@ -1275,7 +1275,7 @@ PHP_FUNCTION(memcache_set_server_params)
 	}
 
 	if (failure_callback != NULL && Z_TYPE_P(failure_callback) != IS_NULL) {
-		if (!IS_CALLABLE(failure_callback, 0, NULL)) {
+		if (!MEMCACHE_IS_CALLABLE(failure_callback, 0, NULL)) {
 			php_error_docref(NULL TSRMLS_CC, E_WARNING, "Invalid failure callback");
 			RETURN_FALSE;
 		}
@@ -1340,7 +1340,7 @@ PHP_FUNCTION(memcache_set_failure_callba
 	}
 
 	if (Z_TYPE_P(failure_callback) != IS_NULL) {
-		if (!IS_CALLABLE(failure_callback, 0, NULL)) {
+		if (!MEMCACHE_IS_CALLABLE(failure_callback, 0, NULL)) {
 			php_error_docref(NULL TSRMLS_CC, E_WARNING, "Invalid failure callback");
 			RETURN_FALSE;
 		}
diff -Naurp memcache-3.0.6/php_memcache.h memcache-3.0.6.oden/php_memcache.h
--- memcache-3.0.6/php_memcache.h	2012-05-04 09:55:52.634821732 +0200
+++ memcache-3.0.6.oden/php_memcache.h	2012-05-04 09:54:51.187411567 +0200
@@ -72,9 +72,9 @@ PHP_FUNCTION(memcache_flush);
 #define MMC_DEFAULT_CACHEDUMP_LIMIT	100		/* number of entries */
 
 #if (PHP_MAJOR_VERSION == 5) && (PHP_MINOR_VERSION >= 3)
-#   define IS_CALLABLE(cb_zv, flags, cb_sp) zend_is_callable((cb_zv), (flags), (cb_sp) TSRMLS_CC)
+#   define MEMCACHE_IS_CALLABLE(cb_zv, flags, cb_sp) zend_is_callable((cb_zv), (flags), (cb_sp) TSRMLS_CC)
 #else
-#   define IS_CALLABLE(cb_zv, flags, cb_sp) zend_is_callable((cb_zv), (flags), (cb_sp))
+#   define MEMCACHE_IS_CALLABLE(cb_zv, flags, cb_sp) zend_is_callable((cb_zv), (flags), (cb_sp))
 #endif
 
 /* internal functions */
